web/src/engine/predictive-text/worker-thread/src/main/correction/distance-modeler.ts

```diff
@@ -350,29 +343,30 @@ export class SearchNode {
           continue;
         }
 
-        calculation = calculation.addInputChar({key: char});
-        calculation = calculation.addMatchChar({key: char, traversal: childTraversal});
+        calculation = calculation.addInputChar(char);
+        calculation = calculation.addMatchChar(char);
       } // else we COULD bundle these as a single subset... but it's likely not that important.
       // after all, the cost ISN'T shifting, so we'll process 'em almost immediately and move on.
 
       keySet.add(char);
```

Comment:
What happens if `char` is falsy? Here and ll.357-358 (where `childTraversal` would equal `undefined`)?

Reply:
So adding an `undefined` charTraversal is valid (not just passing tests, but by design)?

Reply:
Wait, I think I see what you're getting at now.  Let me double-check.

Reply:
OK, yep, good catch - thanks!